integration/tests/neon_evm/test_interoperability.py

```diff
@@ -111,6 +111,79 @@ def test_execute_from_instruction_for_compute_budget(
         resp = solana_caller.execute(COMPUTE_BUDGET_ID, instruction, lamports_amount, sender=sender_with_tokens)
         check_transaction_logs_have_text(solana_client, trx=resp, text="exit_status=0x11")
 
+    def test_execute_with_instruction_struct(
+        self,
+        sender_with_tokens,
+        solana_caller,
+        solana_client,
+    ):
+        instruction_struct = {
```

Comment:
for me it's a bit strange decision to create a custom structure like that and then use decirialization to prepare it for function params serialize_instruction_struct()
More logical to use inner python solana lib class Instruction which we use everywhere 

Also I don't like duplications in execute_with_seed_and_instruction_struct and execute_with_instruction_struct() 
There are just big part of code are the same ias in next exist functions

And  tests itself, old tests could be just parametrised 

But these notes are not critical

Reply:
I replaced custom structure with solana lib `Instruction(...)`. 

We need `serialize_instruction_struct()` method to prepare arg with struct type `ICallSolana.Instruction`, cause we need to pass a tuple (serialized_program_is, serialized_accounts, serialized_data) as the struct. 

We have `execute`, `execute_with_seed`, `execute_with_instruction_struct`, `execute_with_seed_and_instruction_struct` methods. There are duplications in these methods, but they are also parametrized with `lamports` and `is_iterative` options, it already makes them a bit overloaded with logic. I'll add a note about this to the task with refactoring and try to figure out how to improve this part of code.